### PR TITLE
Add a timeout to `Context.GetHTTPClient()`

### PR DESCRIPTION
**Pull request by tamird**
Also DRY acceptance test client creation.

@bram this should make retry actually work when the server is hanging.

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -228,18 +228,10 @@ func TestAcceptEncoding(t *testing.T) {
 	defer leaktest.AfterTest(t)
 	s := StartTestServer(t)
 	defer s.Stop()
-	// We can't use the standard test client. Create our own.
```
Comment (tamird) [inline server/server_test.go]: cc @MBerhault not sure why this was here, but looks like detritus

**Comment by BramGruneir** [comment]
All the changes besides the one Marc should look at LGTM.

File: base/context.go [hunk 2]
```diff
@@ -166,7 +167,10 @@ func (ctx *Context) GetHTTPClient() (*http.Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	ctx.httpClient = &http.Client{Transport: &http.Transport{TLSClientConfig: tlsConfig}}
+	ctx.httpClient = &http.Client{
+		Transport: &http.Transport{TLSClientConfig: tlsConfig},
+		Timeout:   200 * time.Millisecond,
```
Comment (MBerhault) [inline base/context.go]: this is too low. Talking to prod stuff on the other side of the country will get uncomfortably close to it.
It should also be a constant somewhere.

File: server/server_test.go (hunk 1 above)
Comment (MBerhault) [inline server/server_test.go]: odd. apparently Proxy and DisableCompression have no impact on this test. I thought that was the whole point of the test. Oh well, go ahead and remove it then.

File: base/context.go (hunk 2 above)
Comment (tamird) [inline base/context.go]: Fixed and added a constant

File: server/status.go [hunk 3]
```diff
@@ -89,10 +90,6 @@ const (
 	statusStoresPrefix = "/_status/stores/"
 	// statusStorePattern exposes status for a single store.
 	statusStorePattern = "/_status/stores/:store_id"
-
-	// statusProxyTimeout is the timeout used when proxying a request to another
-	// node.
-	statusProxyTimeout = time.Second
```
Comment (tamird) [inline server/status.go]: cc @BramGruneir

File: acceptance/status_server_test.go [hunk 4]
```diff
@@ -123,8 +113,8 @@ func TestStatusServer(t *testing.T) {
 	// Check local response for the every node.
 	for _, node := range l.Nodes {
 		checkNode(t, client, node, idMap[node.ID], "local", idMap[node.ID])
-		get(t, client, node, "/_status/nodes")
-		get(t, client, node, "/_status/stores")
+		get(t, node, "/_status/nodes")
+		get(t, node, "/_status/stores")
```
Comment (MBerhault) [inline acceptance/status_server_test.go]: there are more uses of `client` further down

Comment (tamird) [inline acceptance/status_server_test.go]: yep, removed

File: server/status.go (hunk 3 above)
Comment (BramGruneir) [inline server/status.go]: Even better.

**Comment by MBerhault** [comment]
LGTM

File: server/server_test.go (hunk 1 above)
Comment (bdarnell) [inline server/server_test.go]: Yeah, `DisableCompression` was (part of) the point of the test (I'm not sure why the proxy setting was there). We're no longer testing that we send uncompressed responses when no `Accept-Encoding` was given (go deletes the `Content-Encoding` header after decoding the body, so we can't cleanly tell whether it was ever compressed or not). 

We should still be using `DisableCompression` here, but it's so tricky to make a test that is sensitive to whether or not it was used that it's probably not worth it.

**Pull request merged**